content/2016.09.fullstack-fest-en.rst

```diff
@@ -0,0 +1,195 @@
+
+.. image:: {filename}/images/fullstackfest-vr.jpg
+    :alt: CC-BY @fullstackfest
+    :align: center
+
+Functional programming is in the air. Pure (stateless) functions scale because they can be executed anywere without context. Without side effects, it also becomes obvious to test them. They give backend developers some good pattern for highly available services, and provide the necessary constraints for immutability to frontend developers. I believe that if functional programming is not adopted as much as it could, it is not because of the paradigm and concepts, but because of the disconcerting syntax!
+
+    There is no architecture nirvana!  — Lee Byron
+
+Angular/TypeScript was almost absent from the conference, but we could see a trend around type inference/checking. It really improves developer experience and reduces runtime errors in the end. With projects like Elm, developers feel safe while coding: the compiler is smart and polite, refactoring is fun! The concept of ``null`` is replaced with a notion of *maybe <type>* which makes apps robust.
+
+Everybody seemed to agree that JavaScript transpilation will last for a long time. The language will evolve, and we shouldn't hope for feature freeze. Plus, it really helps developers focus on modern stacks, leveraging modern features like `cancelable promises <https://github.com/tc39/proposal-cancelable-promises>`_. And nowadays it is rather easy to build your app for legacy versions of JavaScript like ES5. But hey, it is a bit absurd that modern browsers have to download big bundles of code full of polyfills! It would be a lot nicer if they were delivered to legacy browsers only — `<!--[if IE 10]>` oh yeah :) — or even better: shipped as something like a browser addon :)
+
+The old times where we could inspect the source code of a page ­— and learn from it — is far! It is sad, but we may have to accept it because WebAssembly goes a lot further and ships code that looks closely to assembler! It's far from the Web we know, but it's still the open Web! Note that it is very different from what we saw in the past with plugins like Java applets, now a WebAssembly function can be called from a usual JS script!
+
+The physical Web is awesome and I'm really excited about this revolution, especially how it could bring a peer-to-peer local Web, using `Web bluetooth <https://developer.mozilla.org/en-US/docs/Web/API/Web_Bluetooth_API>`_ for example. Scott told me that Mozilla approach with `FlyWeb <https://hacks.mozilla.org/2016/09/flyweb-pure-web-cross-device-interaction/>`_ is slightly different because it goes through Wifi, but both teams are in touch. The physical Web is an open project, and Google is pushing it strongly. I'm bothered by the *URL redirections* strategy though beacause it puts the redirection intermadiary in a very strong position. I believe that Google wants to be that intermediary.
```

Comment:
Nit: "Mozilla's approach", or "the Mozilla approach".

"Both teams are in touch" sounds a little bit weird to me but I can't figure out what would sound better.

Typos: "beacause", "intermadiary".